text/00110-compatibility-strategy.md

```diff
@@ -0,0 +1,364 @@
+---
+feature name: compatibility-strategy
+start date: February 2nd, 2020
+rfc pr: https://github.com/aws/aws-cdk-rfcs/pull/111
+related issue: https://github.com/aws/aws-cdk-rfcs/issues/110
+---
+
+<!-- replace the blockquoted sections with your content -->
+# Summary
+
+There are currently two statements we advertise to our customers with regards to framework and cli compatibility:
+
+- CLI versions should always support older framework versions. That is, upgrading the CLI should never require an upgrade of the framework.
+- Framework versions might require an upgrade of the CLI. In which case, we provide instructions as to which CLI version should be used.
+
+This RFC details the necessary strategy and mechanism for ensuring that we keep those promises when releasing new versions.
+
+# Motivation
+
+There are 2 main motivations why we should put some substantial effort into this:
+
+## (1) Haven't we suffered enough?
+
+Recently, we experienced several incidents that are directly related to this domain:
+
+- https://github.com/aws/aws-cdk/issues/5986
+- https://github.com/aws/aws-cdk/issues/4897
+- https://github.com/aws/aws-cdk/issues/4294
```

Comment:
I prefer those formatted as nice links, with more inline context than what you have here, for example mentioning the issue title, so I don't have to click through to know what you're talking about if I'm already aware of those issues:

```md
aws/aws-cdk#5986 - A newer version of the CDK CLI (>= 1.21.0) is necessary to interact with this app
```

And yeah, GitHub will auto-link that issue reference nicely.

Reply:
👍